src/test/java/com/dataartisans/AppFunctionsTests.java

```diff
@@ -0,0 +1,72 @@
+package com.dataartisans;
+
+import com.dataartisans.provided.Event;
+import com.google.common.collect.Lists;
+import org.apache.flink.contrib.streaming.DataStreamUtils;
+import org.apache.flink.streaming.api.TimeCharacteristic;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.RichProcessFunction;
+import org.apache.flink.util.Collector;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for {@link AppFunctions}.
+ */
+public class AppFunctionsTests {
+
+    public static class AssignTimestampsAndWatermarksTests {
+        @Test
+        public void shouldHaveNoTimestampsBeforeAssignment() throws Exception {
+            final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+            env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
+
+            DataStream<Event> stream = env.fromCollection(
+                    new ArrayList<>(Arrays.asList(
+                            new Event(1498404088692L, "some data")
+                    ))
+            );
+
+            DataStream<String> timestampInspectedStream = stream.keyBy("time").process(new TimestampInspector());
```

Comment:
Hmm this `keyBy` is a smell. Perhaps instead of `KeyedStream.process` we want to be using `DataStream.transform`?